micro-rdk/src/common/ota.rs

```diff
@@ -334,71 +327,108 @@ impl<S: OtaMetadataStorage> OtaService<S> {
+                                {
+                                    Ok(pair) => break pair,
+                                    Err(e) => {
+                                        log::error!("failed to build http request: {}", e);
+                                    }
                                 }
                             }
-                        }
-                        Err(e) => {
-                            log::error!("failed to create tcp stream: {}", e);
+                            Err(e) => {
+                                log::error!("failed to create tcp stream: {}", e);
+                            }
                         }
                     }
-                }
-                Err(e) => {
-                    log::error!("failed to create http connection: {}", e);
-                }
-            };
+                    Err(e) => {
+                        log::error!("failed to create http connection: {}", e);
+                    }
+                };
 
-            log::debug!(
-                "retry establishing http connection to {} in {} seconds",
-                &uri,
-                CONN_RETRY_SECS
-            );
-            Timer::after(Duration::from_secs(CONN_RETRY_SECS)).await;
-        };
+                log::debug!(
+                    "retry establishing http connection to {} in {} seconds",
+                    &uri,
+                    CONN_RETRY_SECS
+                );
+                Timer::after(Duration::from_secs(CONN_RETRY_SECS)).await;
+            };
 
-        let conn = Box::new(self.exec.spawn(async move {
-            if let Err(err) = conn.await {
-                log::error!("connection failed: {:?}", err);
+            let conn = Box::new(self.exec.spawn(async move {
+                if let Err(err) = conn.await {
+                    log::error!("connection failed: {:?}", err);
+                }
+            }));
+
+            log::info!("ota connected, beginning download");
+            let request = Request::builder()
+                .method("GET")
+                .uri(uri)
+                .body(Empty::<Bytes>::new())
+                .map_err(|e| OtaError::Other(e.to_string()))?;
+            response = sender
+                .send_request(request)
+                .await
+                .map_err(|e| OtaError::Other(e.to_string()))?;
+
+            // TODO(RSDK-9346): handle other success codes
+            let status = response.status();
+            match (status.is_success(), status.is_redirection()) {
+                (true, false) => break conn,
+                (false, true) => {
+                    log::debug!("target url has been redirected...");
+                    let headers = response.headers();
+                    if !headers.contains_key("location") {
+                        return Err(OtaError::Other(
+                            "`location` not found in redirection response header".to_string(),
```

Comment:
Maybe log the actual response code here so we can understand a little better what went wrong (maybe).